SimPEG/Maps.py

```diff
@@ -20,7 +20,10 @@ def __init__(self, mesh=None, nP=None, **kwargs):
         Utils.setKwargs(self, **kwargs)
 
         if nP is not None:
-            assert type(nP) in integer_types, ' Number of parameters must be an integer.'
+            if type(nP) is np.int64:
+                nP = np.asscalar(nP)
```

Comment:
Just fixed this similarly on another branch. Can we move it into the setter on the `nP`?